### ISSUE TITLE
Epic: Awesome Gitpod Installation and Update/Upgrade Experience

### ISSUE THREAD

**Issue by corneliusludmann** [gitpod-io/gitpod#7315]
## Summary
Gitpod can be self-hosted on-premises. For this, we have recently introduced the Gitpod installer. This epic is about enhancing the Gitpod installer and introducing release management to provide our self-hosted users with an awesome Gitpod installation experience.

## Context
Historically, we offered users [helm charts](https://github.com/gitpod-io/gitpod/tree/63c49622c11389bb379b53d1457ba0d7fa620957/chart) to install Gitpod. Recently, we introduced an [installer](https://github.com/gitpod-io/gitpod/tree/c8af44f6273628f1486f6666920ff497c8ec2322/installer) that makes it easier to install Gitpod. Regular self-hosted releases have not been a high priority for us in the past. That is about to change with this epic.

## Value
For our self-hosted users and (future) customers, it should be as easy as possible to take the first steps towards their own Gitpod installation. They should be reliably and regularly supplied with new Gitpod releases. The update path should be as convenient as possible. It should be easy to upgrade an installation (e.g. unblock enterprise features).

## Acceptance Criteria
- The installer successfully installs Gitpod on all supported platforms and provides a good user experience.
- We provide a way to get installer updates for different channels (like `stable`, `nightly`, etc.). For example by having a Gitpod package management tool (`gpm`).
- We have a release pipeline that releases new Gitpod installer versions (`stable`, `nightly`, etc.).

## Measurement
_(to do)_

## Growth Area
- Onboarding
- Expansion

## In Scope
- Documentation (install, update, upgrade, requirements, etc.)
- Gitpod installer
- Gitpod installer manager (`gpm`) or similar approach
- Gitpod installer release pipeline
- perhaps Gitpod single Docker image

## Out of Scope
- Automated self-hosted tests (it's a separate epic)

**Comment by JonathanTreffler** [comment]
Have you considered offering a prebuilt VM Image for a standard Gitpod Self Hosted setup ?
Many mature open source projects (like Nextcloud) offer these and they are incredibly convenient.
Ideally, there would be a guided Setup (https certificate ...).

Motivation:
I tried installing Gitpod using the helm charts and using the standalone docker before, but have not succeeded yet.
The Installer is presumably already much better, but there are still a few things about the environment you have to figure out yourself to get it working.

I realized Gitpod is very flexible and providing guided installs for all possible self-hosted setups is impossible, but I think it would bring the barrier of entry even lower to have a baseline way of installing Gitpod without anything fancy like multiple worker nodes.

There are probably more important things to improve in the install process, but this could really help people without much experience with Kubernetes.

**Comment by lucasvaltl** [comment]
Marking as this as done because this has been super-seeded by #9929 and #9072
